ocaml/xapi/xapi_xenops.ml

```diff
@@ -1957,6 +1957,7 @@ let update_vusb ~__context (id: (string * string)) =
           let pusb, pusb_r =
             Db.VM.get_VUSBs ~__context ~self:vm
             |> List.map (fun vusb -> Db.VUSB.get_attached ~__context ~self:vusb)
+            |> List.filter (fun attached -> attached <> Ref.null)
```

Comment:
This looks clean but I wonder whether we should call `is_valid_ref` or similar. This would also catch stale references.

Reply:
Right, make sense.